### DISCUSSION TITLE
Smooth animation on price change

### DISCUSSION THREAD

**Discussion by dokocat** [tradingview/lightweight-charts#1392] [Ideas]
Hi! Is there a way to animate the line chart smoothly like this: https://app.deriv.com/ ? The current animation is a bit jumpy, would be nice to know if there was a way to make it smooth like this.

Thanks for any help!

**Comment by stylustrader** [comment]
You have to write your own requestanimationframe plugin

**Reply by dokocat** [reply]
Will the plugin system be available this year?

**Reply by Yaroslav-Bulavin** [reply]
@dokocat Did you find an answer?

**Comment by SlicedSilver** [comment]
This type of smooth animation isn't possible on the library. When an update to the series is provided then it will instantly draw the data point (this isn't animated). The library can animate the movement of the time-scale when it automatically scrolls to keep the new data point in view.

This effect would however be possible with the uses of Plugins but it would require the following:
- A new custom series (in this case a line series which draws the line to the new data point over a few frames).
- and a series primitive for drawing the price line (and the price scale label) in sync with the above custom series. 

Since you will be animating the effect, you would need to tell the library that the plugin would want to draw another frame (this is normally only done when the visible range is changed or a series is updated) by using the [`requestUpdate`](https://tradingview.github.io/lightweight-charts/docs/api/interfaces/SeriesAttachedParameter#requestupdate) callback provided when attaching the primitive to the series.
